ui/components/component-library/text-field/text-field.js

```diff
@@ -18,89 +18,76 @@ import { TextFieldBase } from '../text-field-base';
-      value={value}
-      onChange={handleOnChange}
-      rightAccessory={
-        value && showClear ? (
-          <>
-            {/* replace with ButtonIcon */}
-            <Box
-              className="mm-text-field__button-clear"
-              as="button"
-              display={DISPLAY.FLEX}
-              alignItems={ALIGN_ITEMS.CENTER}
-              justifyContent={JUSTIFY_CONTENT.CENTER}
-              backgroundColor={COLORS.TRANSPARENT}
-              padding={0}
-              {...clearButtonProps} // don't override onClick
-              onClick={handleClear}
-            >
-              <Icon
-                name={ICON_NAMES.CLOSE_OUTLINE}
-                size={SIZES.SM}
-                aria-label="Clear" // TODO: i18n
-                {...clearButtonIconProps}
-              />
-            </Box>
-            {rightAccessory}
-          </>
-        ) : (
-          rightAccessory
-        )
-      }
-      inputProps={{
-        marginRight: showClear ? 6 : 0,
-        ...inputProps,
-      }}
-      {...props}
-    />
-  );
-};
+}) => (
+  <TextFieldBase
+    className={classnames('mm-text-field', className)}
+    value={value}
+    onChange={onChange}
+    rightAccessory={
+      value && showClearButton ? (
+        <>
+          {/* TODO: replace with ButtonIcon */}
+          <Box
+            className="mm-text-field__button-clear"
+            as="button"
+            display={DISPLAY.FLEX}
+            alignItems={ALIGN_ITEMS.CENTER}
+            justifyContent={JUSTIFY_CONTENT.CENTER}
+            backgroundColor={COLORS.TRANSPARENT}
+            padding={0}
+            onClick={clearButtonOnClick}
+            {...clearButtonProps}
+            aria-label="Clear" // TODO: i18n
+          >
+            <Icon name={ICON_NAMES.CLOSE_OUTLINE} size={SIZES.SM} />
+          </Box>
+          {rightAccessory}
+        </>
+      ) : (
+        rightAccessory
+      )
+    }
+    inputProps={{
+      marginRight: showClearButton ? 6 : 0,
+      ...inputProps,
+    }}
+    {...props}
+  />
+);
 
 TextField.propTypes = {
+  /**
+   * The value af the TextField
+   */
+  value: TextFieldBase.propTypes.value.isRequired,
+  /**
+   * The onChange handler af the TextField
+   */
+  onChange: TextFieldBase.propTypes.onChange.isRequired,
   /**
    * An additional className to apply to the text-field
    */
   className: PropTypes.string,
   /**
    * Show a clear button to clear the input
    */
-  showClear: PropTypes.bool,
+  showClearButton: PropTypes.bool,
   /**
-   * The event handler for when the clear button is clicked
+   * The onClick handler for the clear button
    */
-  onClear: PropTypes.func,
+  clearButtonOnClick: PropTypes.func,
   /**
    * The props to pass to the clear button
    */
   clearButtonProps: PropTypes.shape(Box.PropTypes),
-  /**
-   * The props to pass to the icon inside of the close button
-   */
-  clearButtonIconProps: PropTypes.shape(Icon.PropTypes),
```

Comment:
Removing `clearButtonIconProps` because this won't exist once the placeholder clear button is updated with `ButtonIcon`